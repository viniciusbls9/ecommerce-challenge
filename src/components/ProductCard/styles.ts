import styled, { css } from 'styled-components'

export const ProductCard = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: 38rem;
    height: 100%;
    max-height: 40rem;
    box-shadow: 0px 6px 12px #0000001f;
    border: 1px solid #00000026;
    border-radius: 1.4rem;
    background-color: ${theme.colors.white};

    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    img {
      border-radius: 1.4rem;
    }
  `}
`

export const ProductCardHeader = styled.div`
  width: 100%;
  height: 100%;
  max-height: 202px;
  display: flex;
  justify-content: center;
`

export const Image = styled.img`
  width: 100%;
  max-width: 38rem;
`

export const ProductCardBody = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    text-align: center;
    padding: ${theme.spacings.small};
  `}
`

export const ProductTitle = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    line-height: 1.9rem;
    margin-bottom: ${theme.spacings.xsmall};
  `}
`

export const ProductPrice = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.large};
    font-weight: 700;
    line-height: 1.9rem;
    margin-bottom: ${theme.spacings.xsmall};
  `}
`
