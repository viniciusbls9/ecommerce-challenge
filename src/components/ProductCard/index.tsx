import * as S from './styles'
import Link from 'next/link'
import Button from '../Button'
import theme from 'styles/theme'
import { SingleProductProps } from 'models/product'

const SingleProductCard = ({ id, image, title, price }: SingleProductProps) => {
  return (
    <S.ProductCard>
      <S.ProductCardHeader>
        <S.Image src={image} />
      </S.ProductCardHeader>
      <S.ProductCardBody>
        <S.ProductTitle>{title}</S.ProductTitle>
        <S.ProductPrice>{price}</S.ProductPrice>
        <Link href={`/produto/${id?.toString()}`}>
          <Button
            as="a"
            label="ver produto"
            width="100%"
            maxWidth="302px"
            height="36px"
            background="linear-gradient(90deg, rgb(36, 201, 43), rgb(9, 159, 15))"
            backgroundHover="linear-gradient(rgb(9, 159, 15), rgb(36, 201, 43))"
            color={theme.colors.white}
            fontSize={theme.font.sizes.medium}
            boxShadow={`0px 3px 6px ${theme.colors.boxShadow}`}
            margin="0"
          />
        </Link>
      </S.ProductCardBody>
    </S.ProductCard>
  )
}

export default SingleProductCard
