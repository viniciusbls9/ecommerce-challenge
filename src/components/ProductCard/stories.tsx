import { Story, Meta } from '@storybook/react/types-6-0'

import ProductCard from '.'
import { ProductProps } from 'models/product'

export default {
  title: 'ProductCard',
  component: ProductCard
} as Meta

const props = [
  {
    id: 1,
    image: 'https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg',
    title: 'Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops',
    price: 'R$ 109.95'
  }
]

export const Default: Story<ProductProps> = () => (
  <div>
    {props.map((items) => (
      <ProductCard
        key={items.id}
        id={items.id}
        image={items.image}
        title={items.title}
        price={items.price}
      />
    ))}
  </div>
)
