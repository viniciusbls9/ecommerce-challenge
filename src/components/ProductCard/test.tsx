import { renderWithTheme } from 'utils/helpers'
import { screen } from '@testing-library/react'

import ProductCard from '.'

const props = [
  {
    id: 1,
    image: 'https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg',
    title: 'Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops',
    price: 109.95
  }
]

describe('<ProductCard />', () => {
  it('should render the productCard', () => {
    renderWithTheme(
      <ProductCard
        key={props[0].id}
        id={props[0].id}
        image={props[0].image}
        title={props[0].title}
        price={props[0].price}
      />
    )

    expect(
      screen.getByText('Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops')
    )
  })
})
