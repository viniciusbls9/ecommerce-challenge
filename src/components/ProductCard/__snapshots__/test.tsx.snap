// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> should render the productCard 1`] = `
.c0 {
  width: 100%;
  max-width: 38rem;
  height: 100%;
  max-height: 40rem;
  box-shadow: 0px 6px 12px #0000001f;
  border: 1px solid #00000026;
  border-radius: 1.4rem;
  background-color: #FAFAFA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 img {
  border-radius: 1.4rem;
}

.c1 {
  width: 100%;
  height: 100%;
  max-height: 202px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  width: 100%;
  max-width: 38rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  padding: 2.4rem;
}

.c4 {
  font-size: 1.4rem;
  line-height: 1.9rem;
  margin-bottom: 1.6rem;
}

.c5 {
  font-size: 1.8rem;
  font-weight: 700;
  line-height: 1.9rem;
  margin-bottom: 1.6rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  max-width: 302px;
  height: 36px;
  background: linear-gradient(90deg,rgb(36,201,43),rgb(9,159,15));
  margin: 0;
  border-radius: 1.8rem;
  box-shadow: 0px 3px 6px #00000029;
  color: #FAFAFA;
  font-size: 1.6rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      class="c2"
      src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
    />
  </div>
  <div
    class="c3"
  >
    <p
      class="c4"
    >
      Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
    </p>
    <p
      class="c5"
    >
      R$ 109.95
    </p>
    <a
      class="c6"
      color="#FAFAFA"
      font-size="1.6rem"
      height="36px"
      label="ver produto"
      width="100%"
    >
      ver produto
    </a>
  </div>
</div>
`
